### ISSUE TITLE
What kind of relationship should unofficial spaces (Reddit, Discord) have to the community?

### ISSUE THREAD

**Issue by tomodachi94** [NixOS/SC-election-2024#34]
### Question

What kind of relationship should unofficial spaces (Reddit, Discord) have to the community? Should they be linked on the homepage? If so, should they be compelled to comply with the Foundation's code of conduct? Should we attempt to convert these communities into official spaces (if so, at what point)?

### Reminder of the Q&A rules

Please adhere to the [Q&A guidelines and rules](https://github.com/NixOS/SC-election-2024/blob/main/doc/qna.md)

**Comment by cafkafk** [comment]
TL;DR:
- The current subreddit needs to work on it's culture before we can responsibly endorse it
- We need to emphasize the technical concentration on matrix if we consider endorsing any Discord server
- Ideally, we create a community section for spaces we find compatible with our values, while not forcing them to be official NixOS spaces.

I think we've seen the Foundation and by extend the "official" community shift too much to a view of the "NixOS Community" being just NixOS/nixpkgs, NixOS/nix, the matrix channels, discourse, and zulip. I think it's a shame when we have veteran contributors calling lix "off topic" or seeing "aux" as a competitor.

The foundation should prioritize official projects, first and foremost, but it should also act as a force fostering a vibrant ecosystem --- of not only official projects --- but also unofficial community ones. There is a certain elitism from people like me who participate mostly on matrix and through more closed channels of direct messages that "those redditors" aren't as real community members as us, and I think that smells a bit of elitism.

I think reddit has a huge culture problem, and I don't think we should direct people there until that is solved on their side. But I also don't think we should force them to follow our rules just to get mentioned. What matters is they're compatible with our values, and create a space that isn't toxic and harmful for its users.

Regarding discord, it's again easy for me to argue against it on the basis of it being a proprietary platform. In no way should the official project ever add a proprietary product and make it load-bearing unless there is exceptional circumstances demanding it. At the same time, we should meet users where they're comfortable, and try to accommodate their wants and needs. The recent discourse thread about opening a new discord server received a bit too much vitriol in my opinion, users should be allowed to organize how they see fit, without us throwing too many stones at them.

The actual reason I'd not want to link discord on the home page is because I fear it would drive users away from the official matrix channels, where there is a lot of highly competent community members willing to help. I think the need for discord in the community comes from a need for a social space, and perhaps if we made that divide clear on the homepage, that matrix had a more technical slant, then it would be more reasonable. In this case, I also think we shouldn't demand they adopt our policies, but we should ensure that their spaces are compatible with our values. And we should remove them from the homepage in the future if they fail to be compatible with our goals for the wider nix community.

We also need to consider whether to just endorse a single subreddit/discord, or if we should just provide multiple options. A solution is to maintain a list of unofficial community spaces that are compatible, but not necessarily official, and put all of those together on a section on the homepage.

**Comment by mschwaig** [comment]
Unofficial spaces should remain unofficial, the Code of Conduct should therefore of course not apply to them, and they should not be linked anywhere.

How people behave in unofficial spaces is up to them and those spaces, and they should decide as a community which direction they want to take. If their moderators want to apply the CoC, they should feel free to do so.

The official spaces should feel safe to a vast majority of the community, but some members of the community might feel more at home in unofficial spaces that are more ore less protected than the official ones, which is fine too.

I'm saying 'vast majority of the community' and not 'everyone', because abusive behavior is not welcome in official spaces and unfortunately sometimes it takes the form of playing the victim.

**Comment by getchoo** [comment]
> What kind of relationship should unofficial spaces (Reddit, Discord) have to the community? Should they be linked on the homepage?

I think the status quo is in a decent spot, but could see some improvement. We should encourage and allow unofficial communities, but do some vetting in what exactly we link

> should they be compelled to comply with the Foundation's code of conduct?

I'm not sure if "compelled" is the word for what I believe. To be linked on nixos.org, they should *generally* fall in line with the CoC and [our stated values](https://github.com/NixOS/nix-constitutional-assembly/blob/00b53920442a5077fa8c295a9955042e45080323/values.md) -- specifically in basic respect and civility encouraged throughout the community, as well as putting people first. For those that don't, I'm not sure if it would be effective or worth our time to try and force it; the best thing to do in those cases IMO would be to not link to them

>  Should we attempt to convert these communities into official spaces (if so, at what point)?

We should leave the door open if these communities wish to become official spaces, and ensure they meet the standards of our community before accepting them. Attempting to proactively convert them may be more of a disservice to those communities than anything, and create a lot of (pointless) conflict with those who don't share our values

**Comment by numinit** [comment]
> What kind of relationship should unofficial spaces (Reddit, Discord) have to the community?

The internet is a huge place, and as the userbase of NixOS grows, people will form communities for smaller focus groups or projects. We should encourage and support them, even if they have a different focus from us. People have more fun when they find other like-minded people to hang out with and hack on projects they enjoy.

> Should they be linked on the homepage?

Depends on how much space is on the homepage, and whether they can be fairly presented next to other communities of the same size. Long term, it may make less sense for NixOS to be the arbiter of who gets on the homepage if there are very many communities. For now, it seems like it's useful.

> If so, should they be compelled to comply with the Foundation's code of conduct?

No. They can adopt it if they want and there may be benefits in consistency if they do so, but this is a type of conformity that we have no business enforcing on another community. From the values, "we are a synthesis of varied but overlapping communities." We can be both varied and overlapping while treating each other with respect.

> Should we attempt to convert these communities into official spaces (if so, at what point)?

Possibly, but only at the point where the people running those communities think they're ready for it and they agree it would be beneficial. I think we already have a problem with fragmentation of official spaces between Matrix, GH, and the Discourse. Unofficial spaces each bring their own userbases, which is more a feature of online communities than a bug.

**Comment by djacu** [comment]
> What kind of relationship should unofficial spaces (Reddit, Discord) have to the community?

Ideally, a healthy and informal relationship with open communication. Unofficial spaces are excellent ways to grow the user base and sometimes people want to create a new space with like-minded people or for a particular project. We should encourage this.

> Should they be linked on the homepage?

Certainly. We should help people find a happy place to talk and work with other people.

> If so, should they be compelled to comply with the Foundation's code of conduct?

No, that is one part that makes them unofficial. It is there if they want to adopt it but they should be free to create their own CoC.

> Should we attempt to convert these communities into official spaces (if so, at what point)?

We can invite them to become official _if they want to_. Some may see it as mutually beneficial to become official and some may not for a variety of reasons. Both outcomes are fine.

**Comment by winterqt** [comment]
I'm going to be echoing some points here.

> What kind of relationship should unofficial spaces (Reddit, Discord) have to the community?

They should remain unofficial, unless/until they'd like that to change.

> Should they be linked on the homepage? If so, should they be compelled to comply with the Foundation's code of conduct?

Yes, and yes, IMO. However, I also wouldn't use the word "compelled" to describe this; I align with @getchoo's wording here.

> Should we attempt to convert these communities into official spaces (if so, at what point)?

See the first part of my answer.

**Comment by proofconstruction** [comment]
Unofficial spaces like the Reddit and Discord are comfortable contexts for many to engage in, and I have personally participated in discussions on both of these platforms. I don't believe that these platforms should necessarily fall under official purview, or even that they meaningfully *can*. We are in no position to compel compliance, and attempting to do so would just overextend the existing moderation team and send some discussion to other platforms. In the interests of easing the burden on the moderation team, and of maintaining spaces for unrestricted expression, we should leave these platforms alone.

However, I do believe that steps should be taken to improve the experience for participants on the *official* platforms. The unfortunate fact is that when a commercial entity offers space to communicate "for free", the participants are themselves the product (e.g. [OpenAI training models on Reddit posts](https://www.theverge.com/2024/5/16/24158529/reddit-openai-chatgpt-api-access-advertising)). Perhaps worse, a lot of Nix-related technical and procedural knowledge is recorded only in these spaces, which prevents discoverability (especially in the case of the unofficial Discord). Making the official platforms more comfortable will promote their use for a larger portion of the broad community's communications, which will have several positive effects.

**Comment by nyabinary** [comment]
Like many others have said above, I believe unofficial spaces should stay unofficial, but I also would like to see unofficial spaces follow some basic guidelines and have active moderation to be put on the homepage. I also believe that if they want to become official spaces, there should be a pathway to becoming an official space.
